source/Octopus.Cli/Commands/ApiCommand.cs

```diff
@@ -127,6 +129,21 @@ public async Task Execute(string[] commandLineArguments)
             var endpoint = string.IsNullOrWhiteSpace(ApiKey)
                 ? new OctopusServerEndpoint(ServerBaseUrl)
                 : new OctopusServerEndpoint(ServerBaseUrl, ApiKey);
+            
+            /*
+             * There may be a delay between the completion of a large file upload and when Octopus responds
+             * to finish the HTTP connection. This delay can be several minutes. During this time, no traffic is
+             * sent, and some networking infrastructure will close the connection. For example, Azure VMs will
+             * close idle connections after 4 minutes, and AWS VMs will close them after 350 seconds. The
+             * TCP keepalive option will ensure that the connection is not idle at the end of the file upload.
+             *
+             * This is the bug that explains why this doesn't work with .NET Core:
+             * https://github.com/dotnet/corefx/issues/26013
+             */
+            if (keepAlive > 0)
+            {
+                ServicePointManager.FindServicePoint(new Uri(ServerBaseUrl)).SetTcpKeepAlive(true, keepAlive, keepAlive);
```

Comment:
Is it safe to be setting both `keepAliveTime` and `keepAliveTimeInterval` to the same value?
Wouldn't you want `keepAliveTime` to be greater the `keepAliveTimeInterval`?

Reply:
Also, according to the API ( https://docs.microsoft.com/en-us/dotnet/api/system.net.servicepointmanager.settcpkeepalive) both `keepAliveTime` and `keepAliveTimeInterval` is set in milliseconds, but in https://github.com/OctopusDeploy/OctopusCLI/pull/11/files#diff-8f769fe53008cdd8986dbd9314ea7c95R74 we saying seconds?

Reply:
`keepAliveTime` is the amount of time to elapse after the last packet was sent before keepalive is enabled, and then `keepAliveInterval` is the recurring time after that which packets are sent. Values in seconds were enough to keep the connection alive, and I didn't see any reason for the the values to be different. Is there a reason why` keepAliveTime` would be greater than `keepAliveTimeInterval`?

Reply:
I just noticed you are  `* 1000`!
Regarding the 2 settings, honestly don't know, hence the reason to ask.

Reply:
For the purpose of keeping the connection alive while idea, I think both values achieve the same thing. Maybe there are more advanced networking scenarios that might benefit from different values, but I couldn't say what they would be.